src/Parallel/GlobalCache.hpp

```diff
@@ -787,8 +814,10 @@ template <typename GlobalCacheTag, typename Function, typename Metavariables,
           typename... Args>
 void mutate(GlobalCache<Metavariables>& cache, Args&&... args) {
   if (cache.get_main_proxy().has_value()) {
-    cache.thisProxy.template mutate<GlobalCacheTag, Function>(
-        std::make_tuple<Args...>(std::forward<Args>(args)...));
+    if constexpr (not GlobalCache<Metavariables>::is_mocked) {
+      cache.thisProxy.template mutate<GlobalCacheTag, Function>(
+          std::make_tuple<Args...>(std::forward<Args>(args)...));
+    }
```

Comment:
So if the main proxy somehow has been set but the cache is mocked, this won't do anything?

Reply:
done